### PR TITLE
[Feature 20] Pet loses health when past deadline

### PR DESCRIPTION
**Pull request by stephaniesaab**
# Description

Implemented the feature in which the pet loses health when the user has uncompleted tasks past their deadline. This commit includes a pastDeadLine method in the Task class and tests for this method. Also, it includes a method in the ToDoList that returns the sum of the points lost due to uncompleted tasks, each task is worth 1-3 points depending on its priority (HIGH is 3 points, MEDIUM is 2 points, LOW is 1 point). The CheckerUC class calls on this method then on the Pet's ouch method. This commit also fixes redundancies with the CompleteTaskUC which also checks if a task is past its deadline.

Fixes # (20)

## Type of change

- [X] Compilation error fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation
- [X] Test cases
- [X] Redundant code removal (fix code smells)
- [ ] Data storage 
- [ ] UI element

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/useCases/CheckerUC.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Reduce the user's points by the given amount and reduce their pet's health by the given amount of points.
+     * @param user The User object that holds the user's pet and points
+     * @param pointsLost the number of points this user has lost
+     */
+     private void reducePetHealth(User user,int pointsLost) {
+         user.getPet().ouch(pointsLost);
+     }
+
+
```
Comment (XinyingXu) [inline src/main/java/useCases/CheckerUC.java:37]: Well documented and this use case is doing its job correctly. We need to implement its test for the future, but it works fine now.

**Review by XinyingXu — APPROVED** [review]

**Review by sing1348 — APPROVED** [review]
Use case is efficient and well documented.

**Review by serenahonekin — CHANGES_REQUESTED** [review]
I just had a quick comment in regards to hardcoding the points. It should be a quick fix but other than that it looks good!

**Review by serenahonekin — APPROVED** [review]
Looks good!

**Pull request merged**